PagingWithNetworkSample/lib/src/main/java/com/android/example/paging/pagingwithnetwork/reddit/paging/SynchronousRemoteStates.kt

```diff
@@ -0,0 +1,141 @@
+package com.android.example.paging.pagingwithnetwork.reddit.paging
+
+import androidx.paging.CombinedLoadStates
+import androidx.paging.LoadState
+import androidx.paging.LoadState.*
+import androidx.paging.LoadStates
+import com.android.example.paging.pagingwithnetwork.reddit.paging.SynchronousRemoteState.*
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.scan
+import kotlin.Error
+
+/**
```

Comment:
Did you have tests from when you were experimenting with this in the paging codebase?

Reply:
We do but it might take some time to adapt: https://android-review.googlesource.com/c/platform/frameworks/support/+/1591652/7/paging/common/src/main/kotlin/androidx/paging/MutableLoadStateCollection.kt

Let me leave this as a follow-up? I have a couple things on the queue I should probably do before migrating those tests

Reply:
SGTM